### ISSUE TITLE
Not possible to run integration tests on OS X

### ISSUE THREAD

**Issue by cachedout** [elastic/beats#10953]
*Disclaimer:* It's totally possible that there is a way to workaround the problem described below that I'm simply not aware of. :)

It appears to me that it is not currently possible to run Beats integration tests on OS X. Any test which invokes `docker-compose` to bring up a container and then depends on host connectivity between the test being executed and a service on the container will fail with an error that the port on which the service is running cannot be reached.

I suspect that we do not see this in the test suite or under Linux due to a limitation in Docker networking under OS, described here:

https://docs.docker.com/v17.09/docker-for-mac/networking/#per-container-ip-addressing-is-not-possible

Specifically, OS X requires that ports be explicitly exposed in the Docker configuration instead of relying on the host OS bridge networking to just do the right thing and connect the host to the container without any fuss.

Here's an simple example which illustrates this limitation:

Ensuring that no test containers are currently running, from `/metricbeat/module/kibana/stat`, run `go test -tags integration`. Note that tests will fail with an error indicating that the test runner cannot connect to TCP/5601 on localhost, which should be the Kibana docker container that the test starts but due to the Docker OS X bug, cannot be reached.

To illustrate a solution, modify `metricbeat/docker-compose.yml` as follows:
```
   kibana:
     build: ./module/kibana/_meta
+    ports:
+      - "5601:5601"
```
Stop the containers which should be running from the first test. *This is important!* Then re-run the test and note that it passes.

This issue is being filed to discuss possible solutions. Here are a few possible points of discussion.

1) We could generate docker-compose.yml files with Makefiles, adding port bindings if running on OS X.

2) Do nothing and just note in our documentation that integration tests must be run on Linux.

3) We could just modify docker-compose files to explicitly define port bindings. This probably won't harm existing tests but it's a little hard to say for sure.

I'm sure there are other workarounds but I am opening this in order to discuss a path forward before hacking up the code.

Thanks!

**Comment by jsoriano** [comment]
@cachedout this is a bit problematic, yes, I actually also use to need to set `*_HOST` variables when running specific tests, and I use Linux.

>  3\. We could just modify docker-compose files to explicitly define port bindings. This probably won't harm existing tests but it's a little hard to say for sure.

As part of https://github.com/elastic/beats/pull/7957 I found this problem while trying to run standalone tests on different operating systems, and I saw this third option you mention as the sanest one and it works in Linux, Mac and Windows.
This PR introduces this change (along many others). I hope to dedicate some more time to this soon so this can be merged.
